### ISSUE TITLE
[WASM] Drop Event not fired

### ISSUE THREAD

**Issue by nor0x** [unoplatform/uno#7333]
### Current behavior

i have added `AllowDrop` to a `Border` element in order to be able to drag from external sources to an Uno web assembly application. In XAML i have the following

```
    <Border Background="Orange"
            AllowDrop="True"
            x:Name="DropTarget" />
```

in my code-behind i'm checking the events
```

DropTarget.DragEnter += (s, e) =>
{
    Console.WriteLine("drag enter");
};
DropTarget.DragOver += (s, e) =>
{
    Console.WriteLine("drag over");
};

DropTarget.DragLeave += (s, e) =>
{
    Console.WriteLine("drag leave");
};
DropTarget.Drop += (s, e) =>
{
    Console.WriteLine("drop");
};

DropTarget.DropCompleted += (s, e) =>
{
    Console.WriteLine("drop");
};
```

Currently the behavior is that every event EXCEPT `Drop` and `DropCompleted` is fired. During the drag operation the 🚫 glyph is shown on the view
![image](https://user-images.githubusercontent.com/3210391/137395836-29678300-d03c-46d2-a616-d62da106287d.png)

### Expected behavior

`Drop` event should be fired so that we can access the dropped data

### How to reproduce it (as minimally and precisely as possible)

here i have a minimal repro where the issue can be checked

[repro.zip](https://github.com/unoplatform/uno/files/7349409/repro.zip)

### Workaround

haven't found a workaround so far - i will update the issue once i find something

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

3.10.11

### Affected platforms

WebAssembly

### IDE

Visual Studio 2022, Visual Studio 2019, Rider macOS, Visual Studio Code

**Comment by Youssef1313** [comment]
@dr1rrb Was this supposed to be fixed in #6855?

**Comment by dr1rrb** [comment]
@nor0x To allow the drop to be fired you not only have to set the `AllowDrop="true"` but also, in the `DragEnter` or `DragOver`, you have to set the [`DragEventArgs.AcceptedOperation`](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.drageventargs.acceptedoperation) (this will also change the glyph to match the accepted operation).
